docs/docs/learn-by-example/1-writing-an-initial-program/README.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: docs-learn-by-example
+title: Writing the initial program
+permalink: /learn-by-example/1-writing-the-initial-program/
+---
+
+# A complete program by example
+
+This section provides a list of sequential posts on **how to write a complete program from scratch using Arrow**.
+
+We will cover a set of usual concepts like asynchrony, concurrency, controlling effects, parallel and sequential computation, thread switching, domain data and error modeling, resource safety, dependency injection, testing and much more.
+
+## 1. Writing the initial program
+
+Along this series of posts we will gradually build a backend for a Rock Band social network 🎸🤘
+
+<img src="/img/learn-by-example/band_playing.gif" alt="Rock band playing" width="800"/>
+
+Let's start by writing an early simple version of our program to start from there. Let's image a `UserDatabase` for the social network so new users can register:
+
+```kotlin
+interface UserDatabase {
+  fun createUser(name: String): UserId
+  fun findUser(userId: UserId): User?
+}
+```
+
+If our backend was a **Resful API**, we would expect it to offer an endpoint to create new users that would rely on this database.
```

Comment:
```suggestion
If our backend was a **RESTful API**, we would expect it to offer an endpoint to create new users that would rely on this database.
```

Reply:
Great, thanks! :+1: I truly never know how to write this one properly 😅 I'll also ask @MaureenElsberry or @jkmcclellan for a lang review once we complete the tech review.